app/managers/network_manager.ts

```diff
@@ -77,9 +77,14 @@ class NetworkManager {
 
     public createClient = async (serverUrl: string, bearerToken?: string) => {
         const config = await this.buildConfig();
-        const {client} = await getOrCreateAPIClient(serverUrl, config, this.clientErrorEventHandler);
+        let client;
+        try {
+            client = await getOrCreateAPIClient(serverUrl, config, this.clientErrorEventHandler);
+        } catch (error) {
+            throw new Error('Can’t find this server. Check spelling and URL format.');
```

Comment:
@enahum, how would I inject intl in this class?

Reply:
I would probably return an identifier here and let the catch in the component use that id with the `<FormattedText>` component or use `intl.formatMessage` instead of injecting it here.

That said, the reason is cause you want this string to be localized as it can be in any of the supported languages based on the device language at this point.

Reply:
@jfrerich this PR has been forgotten :)